derivable.js.flow

```diff
@@ -4,7 +4,7 @@ export interface Derivable<T> {
 
   derive<E>(f: (value: T) => E): Derivable<E>;
 
-  maybeDerive<E>(f: $NonMaybeType<T> => E): Derivable<E>;
+  maybeDerive<E>(f: $NonMaybeType<T> => E): Derivable<?E>;
```

Comment:
Shouldn't `E` already be nullable if `f` returns a `nullable` value?

Reply:
The problem here is null propagating. I can't make it `Derivable<T | E>` because T may have different type. And flow does not let to extract null from T, so the safest way is making return value potentially nullable.

Reply:
Got it, `maybeDerive` returns `null` if `T` is null. Yeah, that's my bad.